components/brave_sync/brave_sync_service_impl.cc

```diff
@@ -496,6 +499,11 @@ void BraveSyncServiceImpl::OnResolvedPreferences(const RecordsList& records) {
     ResetSyncInternal();
   if (contains_only_one_device)
     OnResetSync();
+
+  if (waiting_for_second_device && !sync_devices->have_less_than_two()) {
```

Comment:
I don't understand why we need this second check and the extra var. Why do we care what the previous size of the sync_devices was? It's also possible that both `this_device_deleted` and `contains_only_one_device` can both be true. The logic here just doesn't make sense to me

Reply:
couldn't we just do something like this?
```
diff --git a/components/brave_sync/brave_sync_service_impl.cc b/components/brave_sync/brave_sync_service_impl.cc
index cb76609e..6fa11216 100644
--- a/components/brave_sync/brave_sync_service_impl.cc
+++ b/components/brave_sync/brave_sync_service_impl.cc
@@ -464,9 +464,8 @@ BraveSyncServiceImpl::PrepareResolvedPreferences(const RecordsList& records) {
 }
 
 void BraveSyncServiceImpl::OnResolvedPreferences(const RecordsList& records) {
-  const std::string this_device_id = sync_prefs_->GetThisDeviceId();
   bool this_device_deleted = false;
-  bool contains_only_one_device = false;
+  bool this_device_added = false;
 
   auto sync_devices = sync_prefs_->GetSyncDevices();
   for (const auto &record : records) {
@@ -480,22 +479,23 @@ void BraveSyncServiceImpl::OnResolvedPreferences(const RecordsList& records) {
           record->syncTimestamp.ToJsTime()),
           record->action,
           &actually_merged);
-      this_device_deleted = this_device_deleted ||
-        (record->deviceId == this_device_id &&
-          record->action == jslib::SyncRecord::Action::A_DELETE &&
-          actually_merged);
-      contains_only_one_device = sync_devices->size() < 2 &&
-        record->action == jslib::SyncRecord::Action::A_DELETE &&
-          actually_merged;
+      if (record->deviceId == sync_prefs_->GetThisDeviceId() &&
+          actually_merged) {
+        this_device_deleted =
+            record->action == jslib::SyncRecord::Action::A_DELETE;
+        this_device_added =
+            record->action == jslib::SyncRecord::Action::A_CREATE;
+      }
     }
   } // for each device
 
   sync_prefs_->SetSyncDevices(*sync_devices);
 
-  if (this_device_deleted)
+  if (sync_devices->size() == 0)
     ResetSyncInternal();
-  if (contains_only_one_device)
-    OnResetSync();
+  else if (sync_devices->size() == 1) {
+    StartLoop(kCheckInitialUpdatesIntervalSec);
+  }
 }
 
 void BraveSyncServiceImpl::OnSyncPrefsChanged(const std::string& pref) {
```